### PR TITLE
Allow goal tolerances aware objects to propagate updates to other objects.

### PR DESCRIPTION
**Pull request by jasonimercer**
Made move_base a goal_tolerances aware object and connected it's goal_tolerances with planners and trackers.

@skaynama  @p6chen

### CODE REVIEW

**Comment by jasonimercer** [comment]
@servos This PR is a PR into the base PR. Adds smarts to the goal tolerances aware object so tolerances can propagate automatically.

**Comment by servos** [comment]
So I understand what this is doing now, however, I'm not sure why we would do this instead of just passing a pointer to an object that everything shares (which I think is what you had before)

**Comment by jasonimercer** [comment]
Yes, I was just handing the pointer around before but once I worked in cpr_pathtracker I saw that that information, in some cases, needs to propagate several levels and would require the same code written over and over. This automates that process and allows any object that is decorated with the GoalTolerancesAware to be able to have goal tolerances and update deeper levels automatically.

**Comment by servos** [comment]
After talking with @jasonimercer this makes sense to me now. 

LGTM 👍

File: nav_core/include/nav_core/goal_tolerances_aware.h [hunk 1]
```diff
@@ -73,8 +121,101 @@ class GoalTolerancesAware
...
+  }
+
+  /**
+   * @brief Convenience method to use the callback mechanism to propagate new goal tolerances
+   *        to other GoalTolerancesAware objects. This version allows raw pointers where the
+   *        object will live until manually deleted.
+   * @param object Object that will get new goal tolerances set
+   */
+  template <typename T>
+  void propagateGoalTolerancesTo(T* object)
```
Comment (skaynama) [inline nav_core/include/nav_core/goal_tolerances_aware.h]: Is this version even used anywhere? I believe all pointers where this function (or rather the other version of it) is used are shared pointers. Or did I miss something and we do actually have a raw pointer lying around?

Comment (jasonimercer) [inline nav_core/include/nav_core/goal_tolerances_aware.h]: https://bitbucket.org/clearpathrobotics/cpr_pathtracker/pull-requests/440/use-goal-tolerances-from-lower-layers-in/diff#Lsrc/vehicle_control_mpc.cppT26

**Comment by skaynama** [comment]
lgtm

**Pull request merged**